homework-12/homework-12.3/homework-12.3/ViewController.swift

```diff
@@ -0,0 +1,37 @@
+//
+//  ViewController.swift
+//  homework-12.3
+//
+//  Created by Kirill Skrytskii on 18.04.22.
+//
+
+import UIKit
+
+class ViewController: UIViewController {
+    
+    @IBOutlet weak var ClickView: UIView!
+    @IBOutlet weak var click: UIButton!
+    
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        click.frame.size = CGSize(width: 60, height: 60)
+        click.layer.cornerRadius = click.frame.width / 2
+        click.backgroundColor = .red
+        
+        addTapGesture()
+}
+
+private func addTapGesture() {
+    let tapGesture = UITapGestureRecognizer(target: self,
+                                            action: #selector(handleLabelTap))
+    
+    click.addGestureRecognizer(tapGesture)
```

Comment:
You shouldn't use `addGestureRecognizer` to add tap action for a button. You should use `addTarget`

Reply:
why can we use addGestureRecognizer ?

Reply:
Because if you add a tap gesture to a button it won't work. All views that inherent `UIControl` need to use `addTarget` to add action